diff/walking/differ.go

```diff
@@ -126,12 +139,19 @@ func (s *walkingDiff) Compare(ctx context.Context, lower, upper []mount.Mount, o
 				if config.Labels == nil {
 					config.Labels = map[string]string{}
 				}
-				config.Labels[uncompressed] = dgstr.Digest().String()
+				config.Labels[LabelUncompressed] = dgstr.Digest().String()
 			} else {
-				if err = archive.WriteDiff(ctx, cw, lowerRoot, upperRoot); err != nil {
+				st, err = archive.WriteDiff(ctx, cw, lowerRoot, upperRoot)
+				if err != nil {
 					return errors.Wrap(err, "failed to write diff")
 				}
 			}
+			if st.Empty {
+				if config.Labels == nil {
```

Comment:
maybe we can put the initialization at the beginning of function.

Reply:
updated